src/chocolatey.console/Program.cs

```diff
@@ -238,20 +238,45 @@ private static void add_assembly_resolver()
 #else
                 var chocolateyPublicKey = ApplicationParameters.UnofficialChocolateyPublicKey;
 #endif
+                if (!requestedAssembly.get_public_key_token().is_equal_to(chocolateyPublicKey))
+                {
+                    // This resolver is only loading official extensions for Chocolatey and ILMerged assemblies.
+                    // Everything else is not handled by this resolver.
+                    return null;
+                }
+
+                // Check if the requested assembly is already loaded
+                var resolvedAssembly = AssemblyResolution.resolve_existing_assembly(requestedAssembly.Name, chocolateyPublicKey);
 
-                if (requestedAssembly.get_public_key_token().is_equal_to(chocolateyPublicKey))
+                if (resolvedAssembly != null)
                 {
-                    // Check if it is already loaded
-                    var resolvedAssembly = AssemblyResolution.resolve_existing_assembly(requestedAssembly.Name, chocolateyPublicKey);
+                    return resolvedAssembly.UnderlyingType;
+                }
 
-                    if (resolvedAssembly != null)
+                if (requestedAssembly.Name.is_equal_to(ApplicationParameters.LicensedChocolateyAssemblySimpleName))
+                {
+                    try
                     {
-                        return resolvedAssembly.UnderlyingType;
+                        "chocolatey".Log().Debug(() => "Resolving reference to chocolatey.licensed...");
+
+                        return AssemblyResolution.load_assembly(
+                            ApplicationParameters.LicensedChocolateyAssemblySimpleName,
+                            ApplicationParameters.LicensedAssemblyLocation,
+                            requestedAssembly.get_public_key_token()).UnderlyingType;
                     }
+                    catch (Exception ex)
+                    {
+                        "chocolatey".Log().Warn("Unable to load chocolatey.licensed assembly. {0}".format_with(ex.Message));
+                        return null;
+                    }
+                }

```

Comment:
This is a breaking change compared to the dynamic loading introduced in v1.2.0.

You are re-introducing a hard-coded path for the licensed chocolatey assembly, which was one of the introductions of the dynamic loading tried to prevent the need.

This must happen after loading extensions or continuing the check for the assembly if none is found in the hard-coded path.

Reply:
Gotcha. The reason I kept this was I noticed that the LicensedAssemblyLocation actually checks two different locations, one of which matches the usual path in /extensions, the other just looks in the same dir as the chocolatey executable.

I'll remove the return here so it can continue into the "normal" dynamic loading.

Reply:
I opted to move this to after the dynamic check instead, retaining the return. That way in most cases the dynamic check should take care of it, and it can be a safe fallback that can return null if it also can't load the licensed extension.

Reply:
> Gotcha. The reason I kept this was I noticed that the LicensedAssemblyLocation actually checks two different locations, one of which matches the usual path in /extensions, the other just looks in the same dir as the chocolatey executable.

Not sure why it checks those two places. If the assembly is located next to the `choco` executable (or in the same directory at least), we wouldn't actually have any control of its loading as it will be loaded automatically by the default assembly loader in almost all cases (not overridable).

Reply:
That's a fair point actually. So this truly seems like it's redundant and not doing anything that the other path isn't already doing, then.

Reply:
@AdmiringWorm it sounds like we're in agreement here, but just to confirm -- I think we should remove this alternate code path for the licensed extension and let it load with the more generic code, yeah?

Reply:
Yeah, I believe we are and that it can be removed.

Unless I have missed something, I believe we will never get into this alternative code path because of the dynamic loading.